embeddings/evaluator/sequence_tagging_evaluator.py

```diff
@@ -1,13 +1,55 @@
-from typing import List, Tuple, Dict, Any
+from typing import List, Tuple, Dict, Any, Optional, Union
 
+import numpy as np
 from datasets import Metric, load_metric
 
 from embeddings.evaluator.metrics_evaluator import MetricsEvaluator
+from itertools import chain
 
 
-class SequenceTaggingEvaluator(MetricsEvaluator):
+class BILOUSequenceTaggingEvaluator(MetricsEvaluator):
+    ALLOWED_TAG_PREFIXES = {"B-", "I-", "L-", "O", "U-"}
+
+    def __init__(self, seqeval_args: Optional[Dict[str, Any]] = None):
+        super().__init__()
+        self.seqeval_args = seqeval_args if seqeval_args else self._get_seqeval_kwargs()
+
+    def _get_seqeval_kwargs(self) -> Dict[str, Any]:
+        return {"suffix": None, "scheme": "BILOU", "mode": "strict"}
+
     @property
     def metrics(self) -> List[Tuple[Metric, Dict[str, Any]]]:
         return [
-            (load_metric("seqeval"), {}),
+            (load_metric("seqeval"), self.seqeval_args),
         ]
+
+    @staticmethod
+    def _convert_single_tag_to_bilou_scheme(
+        input_arr: List[Union[List[str], np.ndarray]]
+    ) -> List[List[str]]:
+        return [[f"U-{tag}" for tag in tags] for tags in input_arr]
+
+    def preprocess_tags(
+        self, input_arr: List[Union[List[str], np.ndarray]]
+    ) -> List[Union[List[str], np.ndarray]]:
+        tags = set(chain.from_iterable(input_arr))
+        tag_prefixes = set(it[0:2] for it in tags)
```

Comment:
`set(tag[0:2] for tag in tags)`?

Reply:
The purpose is to get unique prefixes from tags.